### PR TITLE
feat: Add "requested review" handling + check + refactoring

### LINKED ISSUE #39: purebot require reviews from *all* reviewers instead of just one
Seems like approved PRs are stacking up, after looking at a couple that had been sitting with an `approved` label for hours I found that while the PR had been approved by someone, the person who opened the PR requested reviews from multiple people, and so apparently it's now required that everyone marked as a reviewer needs to approve the changes.

Is this by design?

### PR DESCRIPTION
**Pull request by rhuss**
- If a review is requested the label "pure-bot/requested-review" is added
  (must be configured in the configuration otherwise this functionality is switched off)
- If the last reviewer has been removed and no review has given yet, this
  label is removed (if given)
- A new check "pure-bot/requested-review" check has been added
  * passes if the label "status/review-requested" is not given
  * passes if the label "status/review-requested" is given and at least one reviewer has submitted a review
- The old "pure-bot/wip" check has reverted to the old behaviour
- The internal architecture has been refactored
- A --debug flag has been added for better logging

This reverts #34 and fixes #39

### CODE REVIEW

**Comment by rhuss** [comment]
@zregvart @chirino fancy for a quick review ? If not, no problem. I then would merge and try it out live. 

btw, the bot has been tested on https://github.com/syndesisio/pure-bot-sandbox so far and is still active there.

File: pkg/webhook/add_label_on_review_approval.go [hunk 1]
```diff
@@ -17,42 +17,35 @@ package webhook
...
+	"go.uber.org/zap"
 )
 
-var addLabelOnReviewApprovalHandler Handler = &addLabelOnReviewApproval{}
-
 const (
-	approvedLabel       = "approved"
-	approvedReviewState = "approved"
+	approvedLabel  = "approved"
+	assignedAction = "approved"
```
Comment (zregvart) [inline pkg/webhook/add_label_on_review_approval.go]: This looks a bit fuzzy. See bellow.

File: pkg/webhook/add_label_on_review_approval.go [hunk 2]
```diff
@@ -17,42 +17,35 @@ package webhook
...
+}
+
+func (h *addLabelOnReviewApproval) HandleEvent(eventObject interface{}, gh *github.Client, config config.GitHubAppConfig, logger *zap.Logger) error {
 	event, ok := eventObject.(*github.PullRequestReviewEvent)
 	if !ok {
 		return errors.New("wrong event eventObject type")
 	}
 
-	if event.Installation == nil {
+	if strings.ToLower(event.Review.GetState()) != assignedAction {
```
Comment (zregvart) [inline pkg/webhook/add_label_on_review_approval.go]: Are we checking if the PR has been approved or assigned to someone?

**Review by zregvart — COMMENTED** [review]
Small nitpick, it's not 100% clear from the code what the intention is.

File: pkg/webhook/add_label_on_review_approval.go (hunk 1 above)
Comment (rhuss) [reply pkg/webhook/add_label_on_review_approval.go]: good point, as I havent touched that functionality at all.

I'm blaming IDEA refactoring for now (much better than blaming myself ;-)

going to revert this ...

File: pkg/webhook/add_label_on_review_approval.go (hunk 2 above)
Comment (rhuss) [reply pkg/webhook/add_label_on_review_approval.go]: It's all about approval

**Review by zregvart — APPROVED** [review]
:+1: :robot:

**Pull request merged**